tools/unirestore/scylla_restore.py

```diff
@@ -0,0 +1,996 @@
+                sys.exit(1)
+    logger.debug('Read schema from schema dump: %s', str(keyspaces))
+    return keyspaces
+
+def get_destination_dirs(schema, config, ssh_auth):
+    '''
+    Access the system tables in the destination cluster and gather the UUIDs of the table directories.
+    These will be used to determine the destination paths for the restore files.
+
+    This function also makes sure that the restored schema items are present in the destination cluster.
+
+    Args:
+      schema (dict): schema containing the keyspaces and tables for the restored dataset
+      config (dict): configuration values dictionary
+      ssh_auth (dict): dictionary containing ssh authentication parameters
+
+    Returns:
+      dest_dirs (list): list of destination paths for the restored schema
+    '''
+    def accumulate(acc, item):
+        # accumulator function for the reduce()
+        return acc + len(item[1])
+    host_string = ("CQLSH_HOST=%s" % config['cql_host'])
+    cmd = [host_string, 'cqlsh']
+    if config['cql_user'] != 'None':
+        cmd.append("-u " + config['cql_user'])
+    if config['cql_password'] != 'None':
+        cmd.append('-p ' + config['cql_password'])
+    cmd.append('-e "SELECT * FROM system_schema.tables ;"')
+    cmd = ' '.join(cmd)
+    _out = run_remote(cmd, [config['dest_host']], ssh_auth)
+    out = _out[config['dest_host']]['stdout']
+    dest_dirs = []
+    for line in out:
+        if len(line) > 0:
+            if line.split()[0] in list(schema.keys()):
+                rec = [x for x in line.split() if x != '|']
+                try:
+                    uuid = rec[17]
+                    uuid = rec[17].replace('-', '')
+                except IndexError as e:
+                    logger.error('Error reading system tables line %s\n%s', str(rec), str(e))
+                    sys.exit(1)
+                d = []
+                d.append(config['scylla_data_path'])
+                d.append(rec[0])
+                d.append(rec[1] + '-' + uuid)
+                dest_dirs.append('/'.join(d))
+    if len(dest_dirs) != reduce(accumulate, schema.items(), 0):
+        logger.error('Schema items missing, please create the destination schema first')
+        sys.exit(1)
+    logger.info('destination dirs: %s', str(dest_dirs))
+    return dest_dirs
+
+def get_toc_file_list(path, schema):
+    '''
+    Scan the restored file list for TOC files and return their list
+
+    Args:
+      path (str): root location of the restored file set
+      schema (dict): restored schema dictionary
+
+    Returns:
+      files (list): list of TOC files in the restore set (full path)
+    '''
+    if path.endswith('/'):
+        path += "**"
+    else:
+        path += "/**"
+    _files = glob.glob(path, recursive=True)
+    logger.debug('files detected unfiltered: %s', str(_files))
+    _files = [f for f in _files if os.path.isfile(f) and f.split('/')[-1].endswith('-TOC.txt')]
```

Comment:
splitting paths on '/' isn't consider very portable (assume someone might decide running this on windows...) 
the preferred way in python3 should be using pathlib (it's also simpler to use is most cases) 
https://docs.python.org/3/library/pathlib.html

Reply:
Frankly I don't expect anyone to run this on Windows, the script uses rsync and other very Linux-y things... I probably should document this will only work on Linux

Reply:
regular commandline `ssh` is also problematic on windows, even with putty and other subpar utilities windows has available. In short it's easier and safer to just limit the tool to Linux. 

Still, pathlib is a good suggestion, I'll patch things up to use it instead of split('/')